clojure-mode.el

```diff
@@ -2264,8 +2264,11 @@ position before the current position."
           (while (< (point) position)
             (clojure-forward-logical-sexp 1)
             (clojure-backward-logical-sexp 1)
-            (push (point) sexp-positions)
-            (clojure-forward-logical-sexp 1))
+            ;; Needed to prevent infinite recursion when there's only 1 form in buffer.
+            (if (eq (point) (car sexp-positions))
```

Comment:
I think for numbers it's better to use `=` instead of `eq`.

Reply:
`(car sexp-positions)` will always return `nil` first time, and `=` throws error in this case. Should I keep `eq` or just add a nil check?

Reply:
I guess it's fine to keep `eq` then.

Reply:
Reverted changes, now checks pass